Units/parser-javascript.r/js-scope-resolution.b/expected.tags

```diff
@@ -0,0 +1,10 @@
+Cls	input.js	/^  var Cls = function() {$/;"	c	line:16	function:doesnt_extend
+Cls	input.js	/^var Cls = function() {$/;"	c	line:1
+assert	input.js	/^  function assert(expr, message) {$/;"	f	line:37	function:main
+doesnt_extend	input.js	/^function doesnt_extend() {$/;"	f	line:14
+dummy1	input.js	/^    this.dummy1 = function() {}$/;"	m	line:18	class:doesnt_extend.Cls
+dummy2	input.js	/^  Cls.prototype.dummy2 = function() {$/;"	m	line:20	class:doesnt_extend.Cls
+extend	input.js	/^function extend() {$/;"	f	line:7
+extended	input.js	/^  Cls.prototype.extended = function() {$/;"	m	line:9	class:Cls
```

Comment:
Very interesting.

"Cls" in the scope field is obviously useful information about "extended".
However, "extend" in which "extended" is assigned is also useful information.

None defines the meaning of "scope" field. I have just recognized that there are two type of scopes: 
source code aspect scope and runtime scope.  lexical scope and dynamic scope are more suitable terms?

For reading and understanding source code, source code aspect scope information is also useful. The 
information can be regenerated with line: and end: fiedls but recording it in ctags is not bad idea.

Some parsers may records runtime scope and the other parsers may records source code aspect scope as the value for scope field. It may be too late to redefine the meaning of scope field. However, I would like to explore this area in the future. 

@b4n, you have developed an IDE. I would like to hear your comments.

One of my idea is introducing new scope named sourceScope that is filled only when if source code aspect scope and runtime scope about a tag is not matched.